### PR TITLE
Upgrade to TChannel 1.2.0

### PR DESCRIPTION
**Pull request by kriskowal**
r @prashantv

### CODE REVIEW

**Comment by kriskowal** [comment]
This disables the trace test because there is no API for this in the upgraded TChannel. We can recover the feature and test it when we introduce a dependency on Jaeger.

File: transport_test.go [hunk 1]
```diff
@@ -230,38 +230,38 @@ func TestGetTransportCallerName(t *testing.T) {
...
-
-		tchan, err := getTransport(opts, encoding.Raw)
-		require.NoError(t, err, "getTransport failed")
-		res, err := tchan.Call(ctx, &transport.Request{Method: "test"})
-		require.NoError(t, err, "transport.Call failed")
-
-		assert.Equal(t, tt.traceEnabled, res.Body[0], "TraceEnabled mismatch")
-	}
-}
+// func TestGetTransportTraceEnabled(t *testing.T) {
```
Comment (breerly) [inline transport_test.go]: Is there a better way to disable the test than by commenting it out? For example, something like a:

```
t.Skip("reason this is being skipping")
```

File: transport/tchannel_test.go [hunk 2]
```diff
@@ -174,7 +174,7 @@ func TestTChannelCallSuccessRaw(t *testing.T) {
 	for _, tt := range tests {
 		var lastSpan uint64
 		testutils.RegisterFunc(svr, "echo", func(ctx context.Context, args *raw.Args) (*raw.Res, error) {
-			assert.False(t, tchannel.CurrentSpan(ctx).TracingEnabled(), "Tracing should be disabled")
+			// assert.False(t, tchannel.CurrentSpan(ctx).TracingEnabled(), "Tracing should be disabled")
```
Comment (breerly) [inline transport/tchannel_test.go]: Seems like this should be removed entirely.

File: glide.yaml [hunk 3]
```diff
@@ -13,34 +13,21 @@ import:
...
   - require
 - package: github.com/thriftrw/thriftrw-go
-  version: master
-  subpackages:
-  - ast
-  - compile
-  - envelope
-  - protocol
-  - wire
+  version: ~0.2.1
```
Comment (breerly) [inline glide.yaml]: We should upgrade yab to the latest thriftrw - let's create a task for that.

File: glide.yaml [hunk 4]
```diff
@@ -13,34 +13,21 @@ import:
...
-  - thrift
+  version: ^1.2.0
 - package: golang.org/x/net
   version: master
   subpackages:
   - context
 - package: gopkg.in/yaml.v2
   version: master
+- package: github.com/opentracing/opentracing-go
+  version: ^1.0.0
```
Comment (breerly) [inline glide.yaml]: Just `^1` is fine.

File: glide.yaml [hunk 5]
```diff
@@ -13,34 +13,21 @@ import:
...
-  - protocol
-  - wire
+  version: ~0.2.1
 - package: github.com/uber-go/atomic
   version: ^1.0.0
 - package: github.com/uber/tchannel-go
-  version: ^1.0.9
-  subpackages:
-  - thrift
+  version: ^1.2.0
```
Comment (breerly) [inline glide.yaml]: Just `^1.2` is fine here.

File: glide.yaml [hunk 6]
```diff
@@ -13,34 +13,21 @@ import:
...
-  subpackages:
-  - lib/go/thrift
 - package: github.com/yarpc/yarpc-go
-  version: master
-  subpackages:
-  - encoding/thrift
-  - transport
-  - transport/http
-  - transport/tchannel
+  version: ~0.2.1
```
Comment (breerly) [inline glide.yaml]: I actually didn't realize yab had a dep on yarpc - let's create a ticket to upgrade to the latest version.

Comment (prashantv) [inline glide.yaml]: Yep only for testing, we do a full integration test with YARPC to make sure yab works against it. We should definitely update to the latest.

File: transport/tchannel_test.go (hunk 2 above)
Comment (prashantv) [inline transport/tchannel_test.go]: I think this will come back in the next PR when we add Jaeger?

File: glide.yaml (hunk 6 above)
Comment (breerly) [inline glide.yaml]: Perhaps we should move to `testImport:`?

Comment (prashantv) [inline glide.yaml]: Yep, it's already there!

File: glide.yaml (hunk 3 above)
Comment (kriskowal) [inline glide.yaml]: #123

**Pull request merged**